src/main/java/org/mockito/internal/stubbing/answers/AnswersWithDelay.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.stubbing.answers;
+
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.mockito.stubbing.ValidableAnswer;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+public class AnswersWithDelay implements Answer<Object>, ValidableAnswer, Serializable {
+  private static final long serialVersionUID = 2177950597971260246L;
+
+  private final long sleepyTime;
+  private final Answer<Object> answer;
+
+  public AnswersWithDelay(final long sleepyTime, final Answer<Object> answer) {
+    this.sleepyTime = sleepyTime;
+    this.answer = answer;
+  }
+
+  @Override
+  public Object answer(final InvocationOnMock invocation) throws Throwable {
+    TimeUnit.MILLISECONDS.sleep(sleepyTime);
```

Comment:
I'm not sure what you mean by an "async version"; can you please elaborate?

Without futher details, I'm guessing the answer to the question is "no", since all state in AnswersWithDelay is final.

Though on further thought, it might be interesting to trigger the return externally via another mock method call, rather then make it wait a static time -- that'd be a feature for another time though...